storage/innobase/srv/srv0start.cc

```diff
@@ -419,12 +419,18 @@ static dberr_t create_log_files(char *logfilename, size_t dirnamelen, lsn_t lsn,
 #endif
 
     fsp_flags_set_encryption(log_space->flags);
-    err = fil_set_encryption(log_space->id, Encryption::AES, nullptr, nullptr);
-    ut_ad(err == DB_SUCCESS);
     if (use_dumped_tablespace_keys && !srv_backup_mode) {
-      xb_insert_tablespace_key(log_space->id, log_space->encryption_key,
-                               log_space->encryption_iv);
+      byte key[Encryption::KEY_LEN];
+      byte iv[Encryption::KEY_LEN];
+
+      bool found = xb_fetch_tablespace_key(log_space->id, key, iv);
+      ut_a(found);
+      err = fil_set_encryption(log_space->id, Encryption::AES, key, iv);
```

Comment:
so this is the unencrypted key right? and this sets only in-mem copy?

Also, we were inserting the key to some hash before? is that not necessary now?

Reply:
correct, `xb_fetch_tablespace_key` reads the key from xtrabackup_keys file.

> Also, we were inserting the key to some hash before? is that not necessary now?
I don't remember a hash, `fil_set_encryption` will set the encryption. We had a hash (a map actually) before for the tablespaces missing encryption. And we would check them again in the end of backup.

Reply:
So the old xb_insert_tablespace_key() isn't necessary?

Reply:
Not anymore, before we were first calling `fil_set_encryption` without key & iv, this would cause InnoDB to generete a new pair of key&iv and encrypt it using the default master key, thus we were required to call `xb_insert_tablespace_key` to insert the new keys to `xtrabackup_keys`. However, we changed the approach to copy the key we already have in `xb_insert_tablespace_key` reutilizing the same key. We now call `fil_set_encryption` passing a pair of key&iv making it not generate a new one.

Reply:
Ok, this much happened with the change of two lines :smile: